test/flint/nmod_mat-test.jl

```diff
@@ -473,6 +473,10 @@ end
   @test g == matrix_space(Z17, 3, 3)([1 2 2; 2 13 12; 2 12 15])
 
   @test_throws ErrorException f^(ZZ(2)^1000)
+
+  # issue Oscar.jl#4590
+  a = matrix(Z17, [2 3])
+  @test_throws ErrorException a^2
```

Comment:
Can we also (or instead) add this to the conformance tests for matrices?

Reply:
This should be done eventually, but we don't have any arithmetic tests there right now, so it feels a bit weird to just add this one.